tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts_api.py

```diff
@@ -36,6 +38,33 @@ def test_fetching_artifact_files(user, wandb_init):
     assert open(file_path).read() == "testing"
 
 
+def test_artifact_download_offline_mode(user, wandb_init, monkeypatch):
+    project = "test"
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        # Create the test file in the temporary directory
+        filename = os.path.join(tempdir, "boom.txt")
+        with open(filename, "w") as f:
+            f.write("testing")
+
+        # Use monkeypatch to set WANDB_MODE
+        monkeypatch.setenv("WANDB_MODE", "offline")
+
+        # Mock the wandb.Api().artifact call to prevent actual network requests
+        with patch("wandb.Api.artifact") as mock_artifact:
+            mock_artifact.return_value = wandb.Artifact("test-artifact", "test-type")
+
+            with wandb_init(entity=user, project=project) as run:
+                artifact = wandb.Artifact("test-artifact", "test-type")
+                artifact.add_file(filename, "test-name")
+                run.log_artifact(artifact, aliases=["sequence"])
+
+            with pytest.raises(
+                RuntimeError, match="Cannot download artifacts in offline mode."
+            ):
+                artifact.download()
+
+
```

Comment:
suggestions(non-blocking):
- Instead of directly managing a `TemporaryDirectory`, use the `tmp_path` fixture that comes with `pytest` out of the box -- one less context manager
- rename `filename` -> `file_path` for clarity
- Use the `mocker` fixture from `pytest-mock` to patch stuff.  It'll do the same thing as `with patch(...)` here, including tear-down/cleanup on exiting the test, without the extra level of nesting. Admittedly this isn't obvious if not already familiar with `pytest-mock`, but see below for example usage here
- Use `Path.write_text()` for simple / one-off file write operations.  This is subjective, but IMO it's more readable, succinct, and idiomatic.


```suggestion
def test_artifact_download_offline_mode(user, wandb_init, monkeypatch, tmp_path, mocker):
    project = "test"

    # Create the test file in the temporary directory
    file_path = tmp_path / "boom.txt"
    file_path.write_text("testing")

    # Use monkeypatch to set WANDB_MODE
    monkeypatch.setenv("WANDB_MODE", "offline")

    # Mock the wandb.Api().artifact call to prevent actual network requests
    mock_api_artifact = mocker.patch("wandb.Api.artifact")
    mock_api_artifact.return_value = wandb.Artifact("test-artifact", "test-type")

    with wandb_init(entity=user, project=project) as run:
        artifact = wandb.Artifact("test-artifact", "test-type")
        artifact.add_file(filename, "test-name")
        run.log_artifact(artifact, aliases=["sequence"])

    with pytest.raises(
        RuntimeError, match="Cannot download artifacts in offline mode."
    ):
        artifact.download()


```

That said, IIUC, I don't think you actually need to patch `wandb.Api.artifact` here -- in fact we probably should avoid it here since this is a system test and we want to test real API calls within the (isolated) test environment